api/src/main/java/io/opencensus/trace/samplers/RateLimitingSampler.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace.samplers;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.internal.Utils;
+import io.opencensus.trace.Sampler;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.SpanId;
+import io.opencensus.trace.TraceId;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Sampler that makes a sampling decision based on specified sample QPS (samples per second).
+ *
+ * <p>We use the elapsed time Z since the coin flip to weight our current coin flip. We choose our
+ * probability function P(Z) such that we get the desired sample QPS.
+ *
+ * <p>Let X be the desired QPS. Let Z be the elapsed time since the last sampling decision in
+ * seconds.
+ *
+ * <p>P(Z) = min(Z * X, 1)
+ */
+@AutoValue
+@Immutable
+abstract class RateLimitingSampler extends Sampler {
+
+  static final double NANOS_PER_SECOND = 1.E+9;
+  private final AtomicLong timer = new AtomicLong(0);
+
+  RateLimitingSampler() {}
+
+  abstract double getSamplesPerSecond();
+
+  /**
+   * Returns a new {@link RateLimitingSampler} with desired sample QPS.
+   *
+   * @param samplesPerSecond The desired number of samples per second.
+   * @return a new {@link RateLimitingSampler}.
+   * @throws IllegalArgumentException if {@code samplesPerSecond} is less than 0.
+   */
+  static RateLimitingSampler create(double samplesPerSecond) {
+    Utils.checkArgument(samplesPerSecond >= 0.0, "samplesPerSecond must be non-negative");
+    return new AutoValue_RateLimitingSampler(samplesPerSecond);
+  }
+
+  long getCurrentNanos() {
+    return System.nanoTime();
+  }
+
+  @Override
+  public boolean shouldSample(
+      @Nullable SpanContext parentContext,
+      @Nullable Boolean hasRemoteParent,
+      TraceId traceId,
+      SpanId spanId,
+      String name,
+      List<Span> parentLinks) {
+    // If the parent is present keep the sampling decision.
+    if (parentContext != null && parentContext.isValid()) {
+      return parentContext.getTraceOptions().isSampled();
+    }
+    // If any parent link is present keep the sampling decision.
+    if (parentLinks != null && !parentLinks.isEmpty()) {
+      boolean oneValidLinkFound = false;
+      for (Span parentLink : parentLinks) {
+        if (parentLink.getContext().isValid()) {
+          oneValidLinkFound = true;
+          if (parentLink.getContext().getTraceOptions().isSampled()) {
+            return true;
+          }
+        }
+      }
+      if (oneValidLinkFound) {
+        return false;
+      }
+    }
```

Comment:
Question is: why don't we propagate the negative decision from the parent in `ProbabilitySampler` the same way? According to `SpanBuilderImpl` it's possible only if the span has a remote parent or the sampler is set on the span level explicitly. Nevertheless, the positive decision is propagated always...

As far as I understand, if you decided to blacklist a certain span, you most likely want all child spans to be blacklisted too. Same as with whitelisting.